gateway/gateway.go

```diff
@@ -49,8 +50,8 @@ func NewGateway(options ...Option) (*http.ServeMux, error) {
 func (g gateway) registerEndpoints() (*http.ServeMux, error) {
 	for prefix, registers := range g.endpoints {
 		gwmux := runtime.NewServeMux(
-			runtime.WithProtoErrorHandler(ProtoMessageErrorHandler),
-			runtime.WithMetadata(MetadataAnnotator),
+			append(g.gatewayMuxOptions, runtime.WithProtoErrorHandler(ProtoMessageErrorHandler),
```

Comment:
should our two default options override user-provided options, or vice-versa?

Reply:
I was considering the same. I did it this way purely because it fit with the `append(slice, item, item)` that my brain prefers. Should I swap them?